ambry-api/src/main/java/com.github.ambry/server/StatsHeader.java

```diff
@@ -15,12 +15,14 @@
 package com.github.ambry.server;
 
 import java.util.List;
+import org.codehaus.jackson.annotate.JsonAutoDetect;
 
 
 /**
  * A model object that contains metadata information about some reported stats. For example, the kind of stats that is
  * being reported, timestamp and etc.
  */
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
 public class StatsHeader {
```

Comment:
Do you want to specify `JsonPropertyOrder` for this class too?

Also, are you sure that we need `Visibility.ANY` for this class and `StatsWrapper`? Why does `StatsSnapshot` not need it (is it because it has public setters for each member variable)?

Reply:
Yeah, will specify the `JsonPropertyOrder` for StatsHeader to ensure the order is deterministic in serialized string.

The Jackson annotation `@JsonAutoDetect` is used to tell Jackson to include properties which are not public, both when reading and writing objects. Hence `fieldVisibility = JsonAutoDetect.Visibility.ANY` is needed for both `StatsHeader` and `StatsWrapper`. 
For `StatsSnapshot`, true, it has both public getter and setter which make `fieldVisibility = JsonAutoDetect.Visibility.ANY` unnecessary.